src/CircularArrays.jl

```diff
@@ -50,6 +69,14 @@ CircularArray(def::T, size) where T = CircularArray(fill(def, size))
 CircularVector(data::AbstractArray{T, 1}) where T = CircularVector{T}(data)
 CircularVector(def::T, size::Int) where T = CircularVector{T}(fill(def, size))
 
+Base.IndexStyle(::Type{CircularArray{T,N,A}}) where {T,N,A} = IndexStyle(A)
```

Comment:
The question came to my why this line is be needed. Wouldn't the issues with CartesianIndex be resolved if the IndexStyle defaulted to `IndexCartesian` for anyhting but CircularVector?

Reply:
I thought this was necessary for speed, although it's entirely possible I have got lost in all the versions now. It causes `eachindex(A)` to prefer linear indexing, which can drop straight through to the underlying memory, whereas iterating `CartesianIndices` involves some multiplications by strides... necessary when the underlying array is transposed, say. 

I've pushed what I had lying around.